crates/network-scanner-net/src/async_io.rs

```diff
@@ -0,0 +1,150 @@
+use std::{
+    num::NonZeroUsize,
+    os::windows::io::{AsRawSocket},
+    sync::Arc,
+};
+
+use polling::{Event, Events};
+use socket2::Socket;
+use tokio::sync::{Mutex};
+
+use crate::tokio_raw_socket::TokioRawSocket;
+
+#[derive(Debug)]
+pub struct AsyncIoRuntime {
+    poller: Arc<polling::Poller>,
+    socket_id: usize,
```

Comment:
suggestion: Use an `AtomicUsize` so that you don’t need to take `&mut AsyncIoRuntime`.